_test/test_algorithms/test_cut.m

```diff
@@ -0,0 +1,364 @@
+classdef test_cut < TestCase
+
+properties
+    FLOAT_TOL = 1e-5;
+
+    old_warn_state;
+
+    sqw_file = '../test_sym_op/test_cut_sqw_sym.sqw';
+    ref_file = 'test_cut_ref_sqw.sqw'
+    sqw_4d;
+end
+
+methods
+
+    function obj = test_cut(~)
+        obj = obj@TestCase('test_cut');
+        obj.sqw_4d = sqw(obj.sqw_file);
+
+        obj.old_warn_state = warning('OFF', 'PIXELDATA:validate_mem_alloc');
+    end
+
+    function delete(obj)
+        warning(obj.old_warn_state);
+    end
+
+    function test_you_can_take_a_cut_from_an_sqw_file(obj)
```

Comment:
Very minor point, and unlikely to affect much. Personally, these test names are excessively verbose and give no advantage by being so long. 

`test_cut_sqw_file`
For later ones
`test_cut_sqw_file_to_file` instead of `test_you_can_take_a_cut_from_an_sqw_file_to_another_sqw_file`

Reply:
I can see where you're coming from for this test. When I'm thinking of tests to write, I try to write them as a list of requirements for whatever I'm testing. Which is where the `test_you_can_...` phrasing comes from.

I disagree that things like `test_cut_sqw_file` and `test_cut_sqw_file_to_file` are better names. They are very non-specific: what exactly are the tests testing for, what assertions are they making? This test is probably a bad example of what I'm preaching, and it's something Nick flagged to me earlier, perhaps `test_cut_from_sqw_file_matches_snapshot`?

Reply:
I see exactly where you're coming from, and respect that choice. I think my difference in philosophy comes from that usually I want the test failure to tell me the details (on a pass, I don't care) and a short-ish (but unique) name so that I can easily match to skip/run explicitly. It's very much a matter of opinion and philosophy.